python/nav/portadmin/snmputils.py

```diff
@@ -760,6 +763,21 @@ def _is_trunk(self, interface):
         state = int(self._query_netbox(self.TRUNKPORTSTATE, interface.ifindex))
         return state in [1, 5]
 
+    def is_dot1x_enabled(self, interface):
+        """Returns True or False based on state of dot1x"""
+        return int(self._query_netbox(
+            self.dot1xPortAuth, interface.ifindex)) == b'\xc0'
```

Comment:
Please use named constants instead of magic bitstrings  :-)

Reply:
E.g. define

```python
DOT1X_AUTHENTICATOR = 0b10000000
DOT1X_SUPPLICANT = 0b01000000
```

And use the `&` operator for bitmasking by converting the bytestring to an integer using `ord()` (if you can verify it's only a single byte, as the spec says)

Reply:
Updated with fixes as requested